apiclient/helpers.go

```diff
@@ -343,3 +345,24 @@ func (c *HTTPclient) GenerateSik(sign, secret []byte) ([]byte, error) {
 	}
 	return arbo.BigIntToBytes(arbo.HashFunctionPoseidon.Len(), hash), nil
 }
+
+// Nullifier returns snark friendly nullifier based on the provided signature,
+// the user secret and electionId. The nullifier is calculated following its
+// definition:
+//
+//	nullifier = poseidon(signature, secret*, sha256(electionId))
+//
+// *The secret is optional.
```

Comment:
nit: the `*` has a special meaning in golang so i'd avoid using it for other meanings. you can simply mention "secret", it will be evident you're referring to the parameter, no need for `*`

what does "optional" mean? parameters are never optional like in other languages

my suggestion

```go
//  nullifier = poseidon(signature, secret, sha256(electionId))
//
// secret can be nil, in that case blah will happen
```